lib/experimental/fonts/font-library/class-wp-rest-font-collections-controller.php

```diff
@@ -95,17 +146,126 @@ public function get_item( $request ) {
 			return $collection;
 		}
 
-		$config   = $collection->get_config();
-		$contents = $collection->get_content();
+		$item = $this->prepare_item_for_response( $collection, $request );
 
-		// If there was an error getting the collection data, return the error.
-		if ( is_wp_error( $contents ) ) {
-			$contents->add_data( array( 'status' => 500 ) );
-			return $contents;
+		if ( is_wp_error( $item ) ) {
+			return $item;
 		}
 
-		$collection_data = array_merge( $config, $contents );
-		return rest_ensure_response( $collection_data );
+		return $item;
+	}
+
+	/*
+	* Prepare a single collection output for response.
+	*
+	* @since 6.5.0
+	*
+	* @param WP_Font_Collection $collection Collection object.
+	* @param WP_REST_Request    $request    Request object.
+	* @return array|WP_Error
+	*/
+	public function prepare_item_for_response( $collection, $request ) {
+		$fields = $this->get_fields_for_response( $request );
+		$item   = array();
+
+		$config_fields = array( 'slug', 'name', 'description' );
+		foreach ( $config_fields as $field ) {
+			if ( in_array( $field, $fields, true ) ) {
+				$item[ $field ] = $collection->$field;
+			}
+		}
+
+		$data_fields = array( 'font_families', 'categories' );
+		if ( in_array( 'font_families', $fields, true ) || in_array( 'categories', $fields, true ) ) {
+			$content = $collection->get_content();
+
+			// If there was an error getting the collection data, return the error.
+			if ( is_wp_error( $content ) ) {
+				$content->add_data( array( 'status' => 500 ) );
+				return $content;
+			}
+
+			foreach ( $data_fields as $field ) {
+				if ( in_array( $field, $fields, true ) ) {
+					$item[ $field ] = $content[ $field ];
+				}
+			}
+		}
+
+		return rest_ensure_response( $item );
```

Comment:
I think we also need to add links, including a prepare_links method for the controller.

```suggestion
		$response = rest_ensure_response( $item );

		if ( rest_is_field_included( '_links', $fields ) ) {
			$links = $this->prepare_links( $collection );
			$response->add_links( $links );
		}

		/**
		 * Filters the collection data for a REST API response.
		 *
		 * @since 6.5.0
		 *
		 * @param WP_REST_Response $response  The response object.
		 * @param WP_Font_Collection  $collection Font Collection object.
		 * @param WP_REST_Request   $request.    Request object.
		 */
		return apply_filters( "rest_prepare_font_collection", $response, $collection, $request );
```

Reply:
added here: https://github.com/WordPress/gutenberg/pull/58222/commits/7372ddce73078a70145d1840121299272ef3723f